packages/saltcorn-data/models/field.ts

```diff
@@ -538,7 +540,7 @@ class Field implements AbstractField {
 
   validate(whole_rec: any): ResultMessage {
     const type = this.is_fkey ? { name: "Key" } : this.type;
-    let readval = null;
+    let readval;
```

Comment:
why this change? null is not the same as undefined.

Reply:
I can't explain why it should be null and I don't know if changing this will break it. But I do know that this code is very fiddly due to different types and different representations coming from the browser.